transaction/tests/ResourceId_unittest.cpp

```diff
@@ -0,0 +1,104 @@
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#include "transaction/ResourceId.hpp"
+
+#include "gtest/gtest.h"
+
+namespace quickstep {
+namespace transaction {
+
+TEST(ResourceIdTest, CheckAccessLevels) {
+  const ResourceId db2(2);
+  const ResourceId db2_rel4(2, 4);
+  const ResourceId db2_rel4_block5(2, 4, 5);
+  const ResourceId db2_rel4_block5_tpl8(2, 4, 5, 8);
+
+  EXPECT_TRUE(db2.isDatabaseAccess());
+  EXPECT_FALSE(db2_rel4.isDatabaseAccess());
+  EXPECT_FALSE(db2_rel4_block5.isDatabaseAccess());
+  EXPECT_FALSE(db2_rel4_block5_tpl8.isDatabaseAccess());
+
+  EXPECT_FALSE(db2.isRelationAccess());
+  EXPECT_TRUE(db2_rel4.isRelationAccess());
+  EXPECT_FALSE(db2_rel4_block5.isRelationAccess());
+  EXPECT_FALSE(db2_rel4_block5_tpl8.isRelationAccess());
+
+  EXPECT_FALSE(db2.isBlockAccess());
+  EXPECT_FALSE(db2_rel4.isBlockAccess());
+  EXPECT_TRUE(db2_rel4_block5.isBlockAccess());
+  EXPECT_FALSE(db2_rel4_block5_tpl8.isBlockAccess());
+
+  EXPECT_FALSE(db2.isTupleAccess());
+  EXPECT_FALSE(db2_rel4.isTupleAccess());
+  EXPECT_FALSE(db2_rel4_block5.isTupleAccess());
+  EXPECT_TRUE(db2_rel4_block5_tpl8.isTupleAccess());
+}
+
+TEST(ResourceIdTest, Equality) {
+  // Create some resources.
+  const ResourceId db1(1);
+  const ResourceId db2(2);
+  const ResourceId db2_rel2(2, 2);
+  const ResourceId db2_rel4(2, 4);
+  const ResourceId db2_rel4_block5(2, 4, 5);
+  const ResourceId db2_rel4_block7(2, 4, 7);
+  const ResourceId db2_rel4_block5_tpl3(2, 4, 5, 7);
+  const ResourceId db2_rel4_block5_tpl8(2, 4, 5, 8);
+
+  // Copy some resource ids.
+  const ResourceId db1_copy(db1);
+  const ResourceId db2_rel2_copy(db2_rel2);
+  const ResourceId db2_rel4_block5_copy(db2_rel4_block5);
+  const ResourceId db2_rel4_block5_tpl8_copy(db2_rel4_block5_tpl8);
+
+  // Self comparison must be equal.
+  EXPECT_EQ(db1, db1);
+  EXPECT_EQ(db2_rel2, db2_rel2);
+  EXPECT_EQ(db2_rel4_block5, db2_rel4_block5);
+  EXPECT_EQ(db2_rel4_block5_tpl3, db2_rel4_block5_tpl3);
+
+  // If resources are different, than it must have different ids.
+  EXPECT_NE(db1, db2);
+  EXPECT_NE(db2_rel2, db2_rel4);
+  EXPECT_NE(db2_rel4_block5, db2_rel4_block7);
+  EXPECT_NE(db2_rel4_block5_tpl3, db2_rel4_block5_tpl8);
+
+  // Comparison with a copy must be equal.
+  EXPECT_EQ(db1_copy, db1);
+  EXPECT_EQ(db2_rel2_copy, db2_rel2);
+  EXPECT_EQ(db2_rel4_block5_copy, db2_rel4_block5);
+  EXPECT_EQ(db2_rel4_block5_tpl8_copy, db2_rel4_block5_tpl8);
+}
+
+TEST(ResourceIdTest, ParentResourceIds) {
+  const ResourceId db2(2);
+  const ResourceId db2_rel2(2, 2);
+  const ResourceId db2_rel2_block2(2, 2, 2);
+  const ResourceId db2_rel2_block3(2, 2, 3);
+  const ResourceId db2_rel2_block2_tuple4(2, 2, 2, 4);
+  const ResourceId db2_rel2_block2_tuple7(2, 2, 2, 7);
```

Comment:
I suggest to use a test class derived by `::testing::Test` to hold all the `ResourceId`s, and then test each functionalities using one fixture.

In addition, you need to test every case for the `ResourceId`s, including valid values, invalid, and partially valid ones.